README.md

```diff
@@ -367,7 +367,7 @@ const reactNodeViews = {
 const editorState = EditorState.create({
   schema,
   // You must add the react if you use
-  // the useNodeViews hook.
+  // the useNodeViews or useNodePos hook.
```

Comment:
Not strictly necessary; you can only use `useNodePos` if you're already using `useNodeViews`, since that's how you register React components as node views haha

Reply:
the `useNodePos` hook doesn't throw if the state field provided by this plugin isn't registered, so i feel like we want to note it somewhere? the reason i ran into the omission of this plugin in the first place was that we were using `useNodePos` and node updates were being silently eaten by ProseMirror because the node view position was off by way of that hook falling back to `0` when the `react()` plugin isn't found in state...

is there a different way you'd articulate the relationship between `useNodePos` and `react()`? or are you saying that as of this branch, using `useNodeViews` without `react()` won't work at all? if that's the case i wonder if we should take a harder line and throw (or something) when we can't find the plugin

Reply:
It won't throw, but it won't work properly (in that, it won't organize the React hierarchy correctly). I wouldn't be opposed to throwing if we can't find the plugin, though, I think that's probably a good idea actually! It should be pretty straightforward to do, since we're defining/have access to the plugin key!

All I was pointing out in my comment is that "using `useNodeViews`" already covers "using `useNodePos`" haha.

Reply:
ahhhh. i understand now!

Reply:
> in that, it won't organize the React hierarchy correctly

do things not render correctly? or do they just present incorrectly in React devtools?

Reply:
Uhhhh both I guess. Specifically, this PR makes it so that React Components of ProseMirror children are actually rendered as children of their parent nodes' React Components, which is important if you want context to flow correctly from parent to child. The issue that we were solving at dskrpt was for a collapsible detail/summary node, where we wanted to use context to pass parent state to the child.

So the DOM is rendered correctly either way, but since React hierarchy matters for context flow, things still don't work properly without the plugin.